fe/fe-core/src/main/java/com/starrocks/clone/ColocateTableBalancer.java

```diff
@@ -217,33 +218,47 @@ private void matchGroup() {
-                                        replicationNum, bucketsSeq);
-                                if (st != TabletStatus.HEALTHY) {
-                                    isGroupStable = false;
-                                    LOG.debug("get unhealthy tablet {} in colocate table. status: {}", tablet.getId(),
-                                            st);
-
-                                    TabletSchedCtx tabletCtx = new TabletSchedCtx(
-                                            TabletSchedCtx.Type.REPAIR,
-                                            db.getId(), tableId, partition.getId(), index.getId(), tablet.getId(),
-                                            System.currentTimeMillis());
-                                    // the tablet status will be set again when being scheduled
-                                    tabletCtx.setTabletStatus(st);
-                                    // using HIGH priority, cause we want to stabilize the colocate group as soon as possible
-                                    tabletCtx.setOrigPriority(Priority.HIGH);
-                                    tabletCtx.setTabletOrderIdx(idx);
-
-                                    AddResult res = tabletScheduler.addTablet(tabletCtx, false /* not force */);
-                                    if (res == AddResult.LIMIT_EXCEED) {
-                                        // tablet in scheduler exceed limit, skip this group and check next one.
-                                        LOG.info("number of scheduling tablets in tablet scheduler"
-                                                + " exceed to limit. stop colocate table check");
-                                        break OUT;
+                                // Tablet has already been scheduled, no need to schedule again
+                                if (!tabletScheduler.containsTablet(tablet.getId())) {
+                                    Set<Long> bucketsSeq = backendBucketsSeq.get(idx);
+                                    Preconditions.checkState(bucketsSeq.size() == replicationNum,
+                                            bucketsSeq.size() + " vs. " + replicationNum);
+                                    TabletStatus st = tablet.getColocateHealthStatus(visibleVersion,
+                                            replicationNum, bucketsSeq);
+                                    if (st != TabletStatus.HEALTHY) {
+                                        isGroupStable = false;
+                                        Priority colocateUnhealthyPrio = Priority.HIGH;
+                                        // We should also check if the tablet is ready to be repaired like
+                                        // `TabletChecker` did. Slightly delay the repair action can avoid unnecessary
+                                        // clone in situation like temporarily restart BE Nodes.
+                                        if (tablet.readyToBeRepaired(colocateUnhealthyPrio)) {
```

Comment:
The tolerate of be down is in the function ColocateTableBalancer.checkBackendAvailable. 
The restart time for be is unrelated to the repair time of tablet. How can this avoid the unnecessary clone when be temporarily restart?

Reply:
If the BE come to alive within the delay, the task won't be scheduled.